src/main/java/com/maddyhome/idea/vim/group/OptionGroup.kt

```diff
@@ -65,6 +127,793 @@ internal class OptionGroup : VimOptionGroupBase(), IjVimOptionGroup {
   }
 }
 
+/* Mapping Vim options to IntelliJ settings
+ *
+ * There is an overlap between some Vim options and IntelliJ settings. Some Vim options such as 'wrap' and 'breakindent'
+ * cannot be implemented in IdeaVim, but must be a feature of the host editor, which will have equivalent settings.
+ * Similarly, IntelliJ has settings for features that also exist in IdeaVim, but with a different implementation (e.g.
+ * IntelliJ has the equivalent of 'scrolloff' et al.) These Vim options can still be implemented by IdeaVim, and mapped
+ * to the IntelliJ Setting values.
+ *
+ * The IntelliJ settings implemented are currently closest to Vim's global-local options. There is a persistent global
+ * value maintained by [EditorSettingsExternalizable], and an initially unset local value in [EditorSettings]. The
+ * global value is used when the local value is unset. The main difference with Vim's global-local is that IntelliJ does
+ * not allow us to "unset" the local value. However, we don't actually care about this - it makes no difference to the
+ * implementation.
+ *
+ * IdeaVim will still keep track of what it thinks the global and local values of these options are, but the
+ * local/effective value is mapped to the IntelliJ setting. The current local value of the Vim option is always reported
+ * as the current local/effective value of the IntelliJ setting, so it never gets out of sync. When setting the Vim
+ * option, IdeaVim will only update the IntelliJ setting if the user explicitly sets it with `:set` or `:setlocal`. It
+ * does not update the IntelliJ setting when setting the Vim defaults. This means that unless the user explicitly opts
+ * in to the Vim option, the current IntelliJ setting is used. Changing the IntelliJ setting through the IDE is always
+ * reflected.
+ *
+ * Normally, Vim updates both local and global values when changing the effective value of an option, and this is still
+ * true for mapped options, although the global value is not mapped to anything. Instead, it is used to provide the
+ * value when initialising a new window. If the user does not explicitly set the Vim option, the global value is still
+ * a default value, and setting the new window's local value to default does not update the IntelliJ setting. But if the
+ * user does explicitly set the Vim option, the global value is used to initialise the new window, and is used to update
+ * the IntelliJ setting. This gives us expected Vim-like behaviour when creating new windows.
+ *
+ * Changing the IntelliJ setting through the IDE is treated like `:setlocal` - it updates the local value, but does not
+ * change the global value, so it does not affect new window initialisation.
```

Comment:
I don't observe this behaviour. If I have an explicit `set nu` in `~/.ideavimrc` and to to the IJ Settings | Editor | General | Appearance -> and untick "Show line numbers", nothing in local buffer changes.
Do you mean, there is some other UI that can change this?

Reply:
IdeaVim will set the local value of the IntelliJ setting, so if the Vim option is set, the local IntelliJ value is also set. This comment then assumes the change from IntelliJ is using the _View | Active Editor_ menu. Any change to the IntelliJ global value is not reflected, because the IntelliJ setting's local value overrides this.

If the Vim option is not set, then the comment assumes that the change can come from the Settings dialog, or from the _View | Active Editor_ menu.

Reply:
Updated comments to be more explicit that this refers to the IDE setting's local values, and how this works with the IDE's global value.